### PR TITLE
Feature tidy handlers

### PR DESCRIPTION
**Pull request by kylebaron**
See #46 

# Summary
- Adds field in `SETUP__:` called `flags`
  - Must be a named list
  - All list items must be character
- `flags` sets logical flags in `dots` for the named columns
- When inheriting from a lookup file, we intelligently merge lists of dots rather than clobber; that way, the flags are propagated from the lookup to the final spec; the user can set a flag within dots of the requesting yaml file to prevent propagation from the lookup
- Add `ys_filter()` function that will slice the list based on evaluation of a test expression 
- Add `ys_rename()` to rename columns
- Add `ys_join()` to join two or more specs togehter
- Refactors `ys_select()` so that it will select and rename
- Add `ys_fill_dots()` as a mechanism for filling in `dots` list with sensible defaults

``` r
library(tidyverse)

library(yspec)

spec <- ys_help$spec()
```

# Check logical from `dots`

``` r
ys_filter(spec, covariate)
#>  name c d unit   short                      source           
#>  AGE  - - years  age                        ysdb_internal.yml
#>  WT   - - kg     weight                     ysdb_internal.yml
#>  CRCL - - ml/min CRCL                       .                
#>  ALB  - - g/dL   albumin                    ysdb_internal.yml
#>  BMI  - - m2/kg  BMI                        ysdb_internal.yml
#>  AAG  - - mg/dL  alpha-1-acid glycoprotein  .                
#>  SCR  - - mg/dL  serum creatinine           .                
#>  AST  - - .      aspartate aminotransferase .                
#>  ALT  - - .      alanine aminotransferase   .                
#>  HT   - - cm     height                     ysdb_internal.yml

ys_filter(spec, time_varying)
#>  name c d unit   short  source           
#>  WT   - - kg     weight ysdb_internal.yml
#>  CRCL - - ml/min CRCL   .

ys_filter(spec, go == TRUE)
#> Warning: no columns were selected when filtering
#> [1] name   c      d      unit   short  source
#> <0 rows> (or 0-length row.names)
```

# Access `col`

``` r
ys_filter(spec, col %in% c("WT", "C", "HT"))
#>  name c d unit short             source           
#>  C    + + .    comment character ysdb_internal.yml
#>  WT   - - kg   weight            ysdb_internal.yml
#>  HT   - - cm   height            ysdb_internal.yml
```

# More cases

``` r
ys_filter(spec, unit != "") 
#>  name c d unit         short                     source           
#>  TIME - - hour         TIME                      .                
#>  AMT  - - mg           dose amount               ysdb_internal.yml
#>  DV   - - micrograms/L dependent variable        ysdb_internal.yml
#>  AGE  - - years        age                       ysdb_internal.yml
#>  WT   - - kg           weight                    ysdb_internal.yml
#>  CRCL - - ml/min       CRCL                      .                
#>  ALB  - - g/dL         albumin                   ysdb_internal.yml
#>  BMI  - - m2/kg        BMI                       ysdb_internal.yml
#>  AAG  - - mg/dL        alpha-1-acid glycoprotein .                
#>  SCR  - - mg/dL        serum creatinine          .                
#>  HT   - - cm           height                    ysdb_internal.yml
#>  TAFD - - hours        time after first dose     .                
#>  TAD  - - hours        time after dose           .                
#>  LDOS - - mg           last dose amount          .

ys_filter(spec, unit == "kg")
#>  name c d unit short  source           
#>  WT   - - kg   weight ysdb_internal.yml

ys_filter(spec, type =="character")
#>  name c d unit short                source           
#>  C    + + .    comment character    ysdb_internal.yml
#>  SUBJ + - .    subject identifier   ysdb_internal.yml
#>  RF   + + .    renal function stage .


ys_filter(spec, covariate)  %>% ys_select(-WT)
#>  name c d unit   short                      source           
#>  AGE  - - years  age                        ysdb_internal.yml
#>  CRCL - - ml/min CRCL                       .                
#>  ALB  - - g/dL   albumin                    ysdb_internal.yml
#>  BMI  - - m2/kg  BMI                        ysdb_internal.yml
#>  AAG  - - mg/dL  alpha-1-acid glycoprotein  .                
#>  SCR  - - mg/dL  serum creatinine           .                
#>  AST  - - .      aspartate aminotransferase .                
#>  ALT  - - .      alanine aminotransferase   .                
#>  HT   - - cm     height                     ysdb_internal.yml

ys_filter(spec, discrete)
#>  name  c d unit short                         source           
#>  C     + + .    comment character             ysdb_internal.yml
#>  SEQ   - + .    SEQ                           .                
#>  EVID  - - .    event ID                      ysdb_internal.yml
#>  CP    - + .    Child-Pugh score              .                
#>  MDV   - - .    MDV                           ysdb_internal.yml
#>  BLQ   - + .    below limit of quantification .                
#>  PHASE - - .    study phase indicator         .                
#>  STUDY - + .    study number                  .                
#>  RF    + + .    renal function stage          .

ys_filter(spec, time_varying)
#>  name c d unit   short  source           
#>  WT   - - kg     weight ysdb_internal.yml
#>  CRCL - - ml/min CRCL   .
```

<sup>Created on 2021-03-13 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

### CODE REVIEW

**Comment by kylebaron** [comment]
[test.pdf](https://github.com/metrumresearchgroup/yspec/files/6123908/test.pdf)

File: tests/testthat/test-filter.R [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+test_that("select column filter - decode", {
+  spec <- ys_help$spec()
+  a <- ys_filter(spec, length(decode) > 1)
+  b <- map(spec, "decode")
+  c <- map_int(b, length)
+  d <- c[c > 1]
+  expect_identical(names(a), names(d))
+})
+
+test_that("select column filter - covariate", {
```
Comment (KatherineKayMRG) [inline tests/testthat/test-filter.R:75]: Is it worth adding a test with two arguments where one should test true and the other false? e.g. can it do map(spec, ~isTRUE(.x$dots$covariate), isFALSE(.x$dots$time-varying)) using something like ys_filter(spec, covariate, !time-varying)

Comment (kylebaron) [reply tests/testthat/test-filter.R:75]: Test added.

**Comment by kylebaron** [comment]
[ys-new.pdf](https://github.com/metrumresearchgroup/yspec/files/6174881/ys-new.pdf)

**Pull request merged**